### PR TITLE
Added logging for 4xx responses

### PR DESCRIPTION
**Pull request by whiskeysierra**

### CODE REVIEW

**Comment by lukasniemeier-zalando** [comment]
:+1:

**Comment by lukasniemeier-zalando** [comment]
Is it worth to announce such logging in the `README`, e.g. under "Features"?

**Comment by whiskeysierra** [comment]
> Is it worth to announce such logging in the README, e.g. under "Features"?

I added a new bullet point, but I feel like we should add some more details. We offer several ways to hook into the process, e.g. problem creation, transformation, logging and fallback.

**Comment by AlexanderYastrebov** [comment]
I think at least it should tell that 4xx and 5xx are logged, what levels are and whether it logs stacktrace.

**Comment by lukasniemeier-zalando** [comment]
:+1:

**Comment by lukasniemeier-zalando** [comment]
:+1:

File: src/main/java/org/zalando/problem/spring/web/advice/AdviceTrait.java [hunk 1]
```diff
@@ -191,6 +188,14 @@ default void log(
         return Optional.empty();
     }
 
+    default ResponseEntity<Problem> fallback(
+            @SuppressWarnings("UnusedParameters") final Throwable throwable,
+            @SuppressWarnings("UnusedParameters") final Problem problem,
+            @SuppressWarnings("UnusedParameters") final NativeWebRequest request,
+            @SuppressWarnings("UnusedParameters") final HttpHeaders headers) {
+        return ResponseEntity.status(NOT_ACCEPTABLE).body(null);
```
Comment (lukasniemeier-zalando) [inline src/main/java/org/zalando/problem/spring/web/advice/AdviceTrait.java]: We are currently investigating whether this fallback works in all servlet container envs - I think there is an issue with closing the response if no body is given.

**Pull request merged**